### PR TITLE
docs: Move `eks:DescribeAddon` permission to minimum set of required permissions

### LINKED ISSUE #1582: Add required `eks:DescribeAddon` permission for EKS downstream cluster with EBS addon disabled
## Related Issues

Ref: [GitHub Issue #48144](https://github.com/rancher/rancher/issues/48144)

## Summary

During our investigation of the referenced issue, we discovered that the `eks:DescribeAddon` permission is always required, even when no addon (such as the EBS CSI Driver) is requested while provisioning an EKS cluster through Rancher 2.10/2.9.

Without this permission, cluster provisioning fails with the following error:

Provisioning Error

### Proposed Changes

* **Update Documentation:**
   Move the `eks:DescribeAddon` permission from the EBS-specific permissions block at [Rancher Manager Documentation](https://ranchermanager.docs.rancher.com/v2.10/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-clusters-from-hosted-kubernetes-providers/eks#ebs-csi-driver-addon-permissions) to the main block related to `EKSPermissions`.

* **Correct JSON Policy Examples:**
   Ensure all example JSON policy configurations are well-formed. Currently, there are missing commas, making the examples invalid.

* **Align IAM Permissions Blocks:**

   - Currently, there are two separate `IAMPermissions` policy references:
     - The **Main block**: Used for selecting a Custom Service Role during provisioning from Rancher UI
     - The **Service Role block**: Default role used by Rancher UI when automatically creating roles and policies.
   - Recommendation: Swap the `Service Role` policies with the one in the Main block for consistency and to match Rancher's default behavior.
image

     **Note:**
     - The **Service Role** is automatically used in Rancher UI by default when users allow Rancher to create the required roles and policies.
     - The **IAMPermissions** section in the Main block is only used when users opt to select a custom Service Role manually.

* **Simplify Policy Blocks:**

   - Consolidate policy definitions into a single "Main" block containing all required permissions (including those for VPC and Service Roles).
   - Keep an additional block for optional EBS-specific permissions.

   Current segregation of policies into multiple blocks (Main, Service Roles, VPC, and EBS) is confusing for users and creates unnecessary complexity.

<details>
  <summary>Minimal IAM Policy (based on existing documentation) containing `eks:DescribeAddons` which works and it's well formed</summary>
  
```json
{
  "Version":"2012-10-17",
  "Statement":[
    {
      "Sid":"EC2Permissions",
      "Effect":"Allow",
      "Action":[
        "ec2:AuthorizeSecurityGroupEgress",
        "ec2:AuthorizeSecurityGroupIngress",
        "ec2:CreateKeyPair",
        "ec2:CreateLaunchTemplate",
        "ec2:CreateLaunchTemplateVersion",
        "ec2:CreateSecurityGroup",
        "ec2:CreateTags",
        "ec2:DeleteKeyPair",
        "ec2:DeleteLaunchTemplate",
        "ec2:DeleteLaunchTemplateVersions",
        "ec2:DeleteSecurityGroup",
        "ec2:DeleteTags",
        "ec2:DescribeAccountAttributes",
        "ec2:DescribeAvailabilityZones",
        "ec2:DescribeImages",
        "ec2:DescribeInternetGateways",
        "ec2:DescribeInstanceTypes",
        "ec2:DescribeKeyPairs",
        "ec2:DescribeLaunchTemplateVersions",
        "ec2:DescribeLaunchTemplates",
        "ec2:DescribeRegions",
        "ec2:DescribeRouteTables",
        "ec2:DescribeSecurityGroups",
        "ec2:DescribeSubnets",
        "ec2:DescribeTags",
        "ec2:DescribeVpcs",
        "ec2:RevokeSecurityGroupEgress",
        "ec2:RevokeSecurityGroupIngress",
        "ec2:RunInstances"
      ],
      "Resource":"*"
    },
    {
      "Sid":"CloudFormationPermissions",
      "Effect":"Allow",
      "Action":[
        "cloudformation:CreateStack",
        "cloudformation:CreateStackSet",
        "cloudformation:DeleteStack",
        "cloudformation:DescribeStackResource",
        "cloudformation:DescribeStackResources",
        "cloudformation:DescribeStacks",
        "cloudformation:ListStackResources",
        "cloudformation:ListStacks"
      ],
      "Resource":"*"
    },
    {
      "Sid":"IAMPermissions",
      "Effect":"Allow",
      "Action":[
        "iam:AddRoleToInstanceProfile",
        "iam:AttachRolePolicy",
        "iam:CreateInstanceProfile",
        "iam:CreateRole",
        "iam:CreateServiceLinkedRole",
        "iam:DeleteInstanceProfile",
        "iam:DeleteRole",
        "iam:DetachRolePolicy",
        "iam:GetInstanceProfile",
        "iam:GetRole",
        "iam:ListAttachedRolePolicies",
        "iam:ListInstanceProfiles",
        "iam:ListInstanceProfilesForRole",
        "iam:ListRoles",
        "iam:ListRoleTags",
        "iam:PassRole",
        "iam:RemoveRoleFromInstanceProfile",
        "iam:TagRole"
      ],
      "Resource":"*"
    },
    {
      "Sid":"KMSPermissions",
      "Effect":"Allow",
      "Action":"kms:ListKeys",
      "Resource":"*"
    },
    {
      "Sid":"EKSPermissions",
      "Effect":"Allow",
      "Action":[
        "eks:CreateCluster",
        "eks:CreateFargateProfile",
        "eks:CreateNodegroup",
        "eks:DeleteCluster",
        "eks:DeleteFargateProfile",
        "eks:DeleteNodegroup",
        "eks:DescribeAddon",
        "eks:DescribeCluster",
        "eks:DescribeFargateProfile",
        "eks:DescribeNodegroup",
        "eks:DescribeUpdate",
        "eks:ListClusters",
        "eks:ListFargateProfiles",
        "eks:ListNodegroups",
        "eks:ListTagsForResource",
        "eks:ListUpdates",
        "eks:TagResource",
        "eks:UntagResource",
        "eks:UpdateClusterConfig",
        "eks:UpdateClusterVersion",
        "eks:UpdateNodegroupConfig",
        "eks:UpdateNodegroupVersion"
      ],
      "Resource":"*"
    },
    {
      "Sid":"VPCPermissions",
      "Effect":"Allow",
      "Action":[
        "ec2:AssociateRouteTable",
        "ec2:AttachInternetGateway",
        "ec2:CreateInternetGateway",
        "ec2:CreateRoute",
        "ec2:CreateRouteTable",
        "ec2:CreateSecurityGroup",
        "ec2:CreateSubnet",
        "ec2:CreateVpc",
        "ec2:DeleteInternetGateway",
        "ec2:DeleteRoute",
        "ec2:DeleteRouteTable",
        "ec2:DeleteSubnet",
        "ec2:DeleteTags",
        "ec2:DeleteVpc",
        "ec2:DescribeVpcs",
        "ec2:DetachInternetGateway",
        "ec2:DisassociateRouteTable",
        "ec2:ModifySubnetAttribute",
        "ec2:ModifyVpcAttribute",
        "ec2:ReplaceRoute"
      ],
      "Resource":"*"
    }
  ]
}
```
</details>

### PR DESCRIPTION
**Pull request by yiannistri**
Fixes #1582 

## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, remember to add a "MERGE ON RELEASE" label and set the proper milestone.

## Description

`eks:DescribeAddon` permission is required as Rancher needs to query the API to determine if the EKS CSI  EBS addon is enabled or not for a cluster. The permission is used when building the upstream state for a cluster.

### CODE REVIEW

File: docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-clusters-from-hosted-kubernetes-providers/eks.md [hunk 1]
```diff
@@ -336,7 +291,6 @@ The following are the required permissions for installing the Amazon EBS CSI Dri
         "eks:AssociateIdentityProviderConfig",
         "eks:CreateAddon",
         "eks:DeleteAddon",
-        "eks:DescribeAddon",
```
Comment (thehejik) [inline docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-clusters-from-hosted-kubernetes-providers/eks.md:339]: Please decide whether to keep this in place or remove all duplicate entries which are already in Main block:
```
"eks:DescribeCluster",
"iam:AttachRolePolicy",
"iam:CreateRole",
"iam:GetRole",
"iam:ListAttachedRolePolicies",
"iam:PassRole",
"iam:TagRole",
```

Comment (yiannistri) [reply docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-clusters-from-hosted-kubernetes-providers/eks.md:339]: Good catch, I will remove the duplicated entries.

**Review by thehejik — DISMISSED** [review]
Thanks a lot! This PR matches our findings and makes the EKS permissions part easy to understand. Admins can just copy & paste to AWS / IAM Policy and assign to given user.

**Review by thehejik — APPROVED** [review]

**Review by sunilarjun — APPROVED** [review]
Thank you for updating and versioning!

**Pull request merged**